app/oidc/oidc.py

```diff
@@ -0,0 +1,66 @@
+import functools
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta, timezone
+from typing import Callable, ParamSpec
+
+import pytz
+from google.auth.credentials import Credentials
+from google.auth.transport.requests import Request
+from structlog import get_logger
+
+from app.settings import OIDC_TOKEN_LEEWAY_IN_SECONDS
+
+P = ParamSpec("P")
+
+logger = get_logger()
+TTL = 3600 - OIDC_TOKEN_LEEWAY_IN_SECONDS  # 1 hour minus leeway in seconds
```

Comment:
For clarity, we should assign 3600 to a constant or add comments. i.e `OIDC_TOKEN_VALIDITY_IN_SECONDS`